tests/test_kafka.py

```diff
@@ -0,0 +1,181 @@
+# Currently using stub-universe so it's beta-kafka, for the last time
+KAFKA_PACKAGE_NAME = "beta-kafka"
+KAFKA_SERVICE_NAME = "secure-kafka" if KERBERIZED_KAFKA else "kafka"
+
+
+@pytest.fixture(scope='module')
+def kerberized_kafka():
+    try:
+        LOGGER.warning('Temporarily using Kafka stub universe until kerberos is released')
+        sdk_cmd.run_cli('package repo add --index=0 {} {}'.format(
+        'kafka-aws',
+        'https://universe-converter.mesosphere.com/transform?url=https://infinity-artifacts.s3.amazonaws.com/beta-kafka/20171116-095256-ui8c625WXxUN6abI/stub-universe-beta-kafka.json'))
+
+        fqdn = "{service_name}.{host_suffix}".format(service_name=KAFKA_SERVICE_NAME,
+                                                     host_suffix=sdk_hosts.AUTOIP_HOST_SUFFIX)
+
+        brokers = ["kafka-0-broker", "kafka-1-broker", "kafka-2-broker"]
+
+        principals = []
+        for b in brokers:
+            principals.append("kafka/{instance}.{domain}@{realm}".format(
+                instance=b,
+                domain=fqdn,
+                realm=sdk_auth.REALM))
+
+        principals.append("client@{realm}".format(realm=sdk_auth.REALM))
+
+        kerberos_env = sdk_auth.KerberosEnvironment()
+        kerberos_env.add_principals(principals)
+        kerberos_env.finalize()
+
+        service_kerberos_options = {
+            "service": {
+                "name": KAFKA_SERVICE_NAME,
+                "security": {
+                    "kerberos": {
+                        "enabled": True,
+                        "kdc_host_name": kerberos_env.get_host(),
+                        "kdc_host_port": int(kerberos_env.get_port()),
+                        "keytab_secret": kerberos_env.get_keytab_path(),
+                    }
+                }
+            }
+        }
+
+        sdk_install.uninstall(KAFKA_PACKAGE_NAME, KAFKA_SERVICE_NAME)
+        sdk_install.install(
+            KAFKA_PACKAGE_NAME,
+            KAFKA_SERVICE_NAME,
+            DEFAULT_KAFKA_TASK_COUNT,
+            additional_options=service_kerberos_options,
+            timeout_seconds=30 * 60)
+
+        yield kerberos_env
+
+    finally:
+        sdk_install.uninstall(KAFKA_PACKAGE_NAME, KAFKA_SERVICE_NAME)
+        if kerberos_env:
+            kerberos_env.cleanup()
+
+
+@pytest.fixture(scope='module', autouse=True)
+def setup_spark(kerberized_kafka):
+    try:
+        utils.require_spark()
+        yield
+    finally:
+        utils.teardown_spark()
+
+
+@pytest.mark.sanity
+@pytest.mark.skipif(not utils.kafka_enabled(), reason='KAFKA_ENABLED is false')
+def test_spark_and_kafka():
+    def producer_launched():
+        return utils.streaming_job_launched(PRODUCER_SERVICE_NAME)
+
+    def producer_started():
+        return utils.streaming_job_running(PRODUCER_SERVICE_NAME)
+
+    def kafka_broker_dns():
+        cmd = "dcos {package_name} --name={service_name} endpoints broker".format(
```

Comment:
nit: you could use this helper function instead: https://github.com/mesosphere/spark-build/blob/master/testing/sdk_cmd.py#L35